src/webgpu/api/operation/rendering/stencil.spec.ts

```diff
@@ -230,3 +230,65 @@ g.test('stencil_compare_func')
     const { stencilCompare, stencilRefValue, _expectedColor } = t.params;
     t.checkStencilCompareFunction(stencilCompare, stencilRefValue, _expectedColor);
   });
+
+g.test('stencil_read_write_mask')
+  .desc(
+    `
+  Tests that setting a stencil read/write masks work. Basically, The base triangle sets 1 to the
+  stencil, and then try to draw a triangle with different stencil values.
+    - If the stencil of the triangle is 0, it does not draw because
+      'base stencil(1) & read mask(1) != triangle stencil(0)'.
+    - If the stencil of the triangle is 1, it draws because
+      'base stencil(1) & read mask(1) == triangle stencil(1)'.
+    - If the stencil of the triangle is 1, it does not draw because
+      'base stencil(1) & read mask(1) == triangle stencil(2)'.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('maskType', ['read', 'write'])
+      .combineWithParams([
+        { stencilRefValue: 0, _expectedColor: kBaseColor },
+        { stencilRefValue: 1, _expectedColor: kStencilColor },
+        { stencilRefValue: 2, _expectedColor: kBaseColor },
+      ])
+  )
+  .fn(async t => {
+    const { maskType, stencilRefValue, _expectedColor } = t.params;
+
+    const depthSpencilFormat: GPUTextureFormat = 'depth24plus-stencil8';
+
+    const baseStencilState = {
+      compare: 'always',
+      failOp: 'keep',
+      passOp: 'replace',
+    } as const;
+
+    const stencilState = {
+      compare: 'equal',
+      failOp: 'keep',
+      passOp: 'keep',
+    } as const;
+
+    const baseState = {
+      format: depthSpencilFormat,
+      depthWriteEnabled: false,
+      depthCompare: 'always',
+      stencilFront: baseStencilState,
+      stencilBack: baseStencilState,
+      stencilReadMask: 0xff,
+      stencilWriteMask: maskType === 'write' ? 0x1 : 0xff,
+    } as const;
+
+    const state = {
+      format: depthSpencilFormat,
+      depthWriteEnabled: false,
+      depthCompare: 'always',
+      stencilFront: stencilState,
+      stencilBack: stencilState,
+      stencilReadMask: maskType === 'read' ? 0x1 : 0xff,
```

Comment:
The test will also still pass if you remove this (simulating a bug where stencilReadMask is ignored). The reason is similar: the stencil attachment is cleared to 1 by runStencilStateTest, so the mask has no effect.

Reply:
ditto.